### PR TITLE
address #9: match boxing with (Group) OSCORE

### PR DESCRIPTION
**Pull request by marco-tiloca-sics**
Match boxing when using OSCORE, KUDOS, and Group OSCORE

### CODE REVIEW

File: draft-bormann-core-corr-clar.md [hunk 1]
```diff
@@ -365,37 +367,41 @@ boxes that also do not fit certain likely use cases of these additions
...
+
+  Furthermore, {{-oscore}} tacitly assumes that the two peers terminate any ongoing CoAP Observation {{-observe}} that they still have ongoing, upon installing a new OSCORE Security Context, irrespective of the method used to perform a key update.
+
+  On these premises, a belated response protected with the old OSCORE Security Context will fail decryption, due to that Security Context not available anymore on the receiving client.
+
+* The recent key update protocol KUDOS {{-kudos}} specifies how two OSCORE peers can establish a new shared OSCORE Security Context that replaces their old one, with a number of advantages compared to the method defined in {{Section B.2 of -oscore}}.
+
+  Like OSCORE, KUDOS does not impact the CoAP matching functions based on Message ID or Token either.
+
+  However, KUDOS allows the two OSCORE peers to agree on preserving their ongoing CoAP Observations across the performed key update. If and only if both peers agree to do that during an execution of KUDOS, will their Observations remain active after installing the new OSCORE Security Context, which the two peers will use from then on to protect their exchanged Observe notifications.
```
Comment (cabo) [inline draft-bormann-core-corr-clar.md]:
```suggestion
  However, KUDOS allows the two OSCORE peers to agree on preserving their ongoing CoAP Observations across the performed key update. If and only if both peers agree to do that during an execution of KUDOS, their Observations will remain active after installing the new OSCORE Security Context, which the two peers will use from then on to protect their exchanged Observe notifications.
```

(this isn't just about observe, but any request with pending responses?)

File: draft-bormann-core-corr-clar.md [hunk 2]
```diff
@@ -365,37 +367,41 @@ boxes that also do not fit certain likely use cases of these additions
...
+
+  Furthermore, irrespective of the method used to perform a key update, {{Section 3 of -kudos}} updates OSCORE and {{-oscore}} in order to prevent security issues that can arise from misbinding a request and a response, when those are protected with two different OSCORE Security Contexts.
+
+  Such an update requires a server to include its own Sender Sequence Number as Partial IV of an outgoing response, when protecting it with a Security Context different from the one used to protect the corresponding request. An exception safely applies to the response messages that are sent when running the key update procedure defined in {{Section B.2 of -oscore}}.
```
Comment (cabo) [inline draft-bormann-core-corr-clar.md]: "Such an update" is the key update, not the update of 8613, no?

File: draft-bormann-core-corr-clar.md [hunk 3]
```diff
@@ -365,37 +367,41 @@ boxes that also do not fit certain likely use cases of these additions
...
-   to be defined; also, a way to define the progression of matching
-   boxes needs to be defined that is compatible with the security
-   properties of the underlying protocols.
-   This may require new efforts, to be initiated based on some
-   formative contributions.
+The match boxing semantics of the current protocol are clearly defined, but can be unsatisfactory given the current requirements.
+
+This calls for careful design choices and enhancements when developing extensions for CoAP or protocols and methods applicable to CoAP, such as in the following cases:
+
+* As originally defined in {{-oscore}}, the security protocol Object Security for Constrained RESTful Environments (OSCORE) does not impact the CoAP matching functions based on Message ID or Token.
```
Comment (cabo) [inline draft-bormann-core-corr-clar.md]: I'm having a hard time with "does not impact".  It sure does, but maybe only in one way and not another?  I think we need to be more precise here (maybe even adding an example).

File: draft-bormann-core-corr-clar.md (hunk 1 above)
Comment (marco-tiloca-sics) [reply draft-bormann-core-corr-clar.md]: Correct.

File: draft-bormann-core-corr-clar.md (hunk 2 above)
Comment (marco-tiloca-sics) [reply draft-bormann-core-corr-clar.md]: It refers to the update made to RFC 8613. Maybe s/Such an update/Such an update to OSCORE

File: draft-bormann-core-corr-clar.md (hunk 3 above)
Comment (marco-tiloca-sics) [reply draft-bormann-core-corr-clar.md]: Largely addressed in https://github.com/core-wg/corrclar/commit/d16eb1af94ed916c6c57e6c07a73d03c2918d5da

**Pull request merged**